extensions/sockets/src/main/java/org/mule/extension/socket/api/source/SocketListener.java

```diff
@@ -80,12 +87,34 @@ public void start() throws Exception {
     executorService =
         newSingleThreadExecutor(r -> new Thread(r,
                                                 format("%s%s.socket.listener", getPrefix(muleContext), flowConstruct.getName())));
+
     stopRequested.set(false);
-    executorService.execute(this::listen);
+    executorService.execute(() -> listen(sourceCallback));
+  }
+
+  @OnSuccess
+  public void onSuccess(@Optional(defaultValue = "#[payload]") @XmlHints(allowReferences = false) Object responseValue,
```

Comment:
not sure... anita had it like that in the original connector. This will go away with the `@Content` thing though so I would worry too much